src/transformers/trainer.py

```diff
@@ -2357,7 +2357,7 @@ def _inner_training_loop(
                     model = self.accelerator.prepare(self.model)
                 else:
                     if delay_optimizer_creation:
```

Comment:
cc @IlyasMoutawwakil as you wanted to remove this 👀

Reply:
you see why I remived it, is because currently we do create the optimizer here, and we need to prepare the fsdp model as well (otherwise it fails), so the two branches of the if statement become the same